cl/phase1/network/services/interface.go

```diff
@@ -0,0 +1,14 @@
+package services
+
+import (
+	"context"
+
+	"github.com/ledgerwatch/erigon/cl/cltypes"
+)
+
+type Service[T any] interface {
```

Comment:
will try if i can

Reply:
and i like to operate on interface, because it makes writing unittests way easier.
I think unittest writability (dont know accurate word to this property) is also a benchmark to evaluate code quality.

Reply:
Well, i guess it is not super-simple. i need to finish off the PR first but I think it is possible

Reply:
sure, I can try to do what I imagined as you finish this part.